specification.md

```diff
@@ -344,3 +337,183 @@ In this case, if the `source` value is `source-abc` and the `booking-referral` i
+}
+```
+
+
+### Data “Freshness”
+
+Aggregators may request or receive information from publishers at different times, and understanding how recently data about a Location, Schedule, or Slot was retrieved can help end users gauge the accuracy of items in an aggregated dataset. _Slot Aggregators_ SHOULD use the `lastSourceSync` extension on the `meta` field of any resource to indicate the last time at which the data was known to be accurate:
+
+```js
+{
+  "resourceType": "Schedule",
+  "id": "123",
+  "meta": {
+    "extension": [{
+      "url": "http://hl7.org/fhir/StructureDefinition/lastSourceSync",
+      "valueDateTime": "2021-04-19T20:35:05.000Z"
+    }]
+  }
+  // additional Schedule fields here...
+}
+```
+
+**Note:** `lastSourceSync` has been tentatively approved, but is not yet finalized as part of FHIR as of April 28, 2021. (You can keep up-to-date on the status of this extension in FHIR’s issue tracker at [FHIR-31567][].)
+
+
+### Unknown Availability, Capacity, or Slot Times
+
+Because source systems may experience errors or may not conform to the SMART Scheduling Links specification, _Slot Aggregators_ need additional tools to describe unusual situations that are not relevant to first-part _Slot Publishers_. Specifically, _Slot Aggregators_ may describe locations with:
+- Unknown slot availability (e.g. because a source system is unreachable),
+- Unknown capacity or times (e.g. slots are known to be free or busy only _at some time_ in the near future).
+
+Features and guidelines for these scenarios SHALL NOT be used by first-party _Slot Publishers_; they are described only for use by _Slot Aggregators_.
+
+
+#### Unknown Availability
+
+If there is no source of appointment data for a Schedule or a source system is unreachable, has errors, or cannot be reliably handled for some reason, a _Slot Aggregator_ should publish a single slot for the Schedule with a `status` value of `"free"` and a “Capacity” extension with a value of `0`. First-party _Slot Publishers_ SHOULD NOT publish a slot with these features; if a slot has no free capacity, the slot should have a `status` value of `"busy"`.
+
+Example usage:
+
+```json
+{
+  "resourceType": "Slot",
+  "id": "789",
+  "schedule": {
+    "reference": "Schedule/456"
+  },
+  "status": "free",
+  "start": "2021-03-10T00:00:00Z",
+  "end": "2021-03-24T00:00:00Z",
+  "extension": [{
+    "url": "http://fhir-registry.smarthealthit.org/StructureDefinition/slot-capacity",
+    "valueInteger": 0
+  }]
+}
+```
+
+
+#### Unknown Capacity or Slot Times
+
+_Slot Aggregators_ should use the **optional "Has Availability" extension** to convey that a Schedule has non-zero future availability, without conveying details about when or how much. When used on a Schedule, that Schedule MAY have no associated Slots. _Slot Publishers_ SHALL NOT use this capacity in place of publishing granular Slots; it is defined to support Slot Aggregators (i.e. systems that re-publish Slot data from other APIs).
```

Comment:
I’ve added a sentence in the middle here about a schedule using this not needing slots. I wrote “MAY have no associated Slots,” but that leaves things a bit ambiguous (you could use it as a shortcut/summary even when slot-level detail is available here). Should we change this to “SHALL” or, conversely, explicitly suggest that this can be used to summarize data in slots?

See also the two comments above:
1. Would it make sense to allow this on slots, too? It would describe the CVS-style case well, and now that this is in a separate section from the rest of the spec, it might run *less* afoul of worries that publishers would abuse it. Not a huge deal either way, as the abuse is definitely a concern.
2. If we allowed a no-value case (indicating unknown availability) or switched it from `valueBoolean` to `valueCode`, that would solve the “unknown availability” case in a pretty elegant way.